src/Stratis.Bitcoin/P2P/Peer/NetworkPeer.cs

```diff
@@ -568,14 +574,18 @@ private void OnStateChanged(NetworkPeerState previous)
                     }
                     catch (TargetInvocationException ex)
                     {
-                        this.TraceCorrelation.LogInside(() => NodeServerTrace.Error("Error while StateChanged event raised", ex.InnerException));
+                        this.logger.LogError("Exception occurred: {0}", ex.InnerException.ToString());
```

Comment:
Here and for all similar occurrences of this log statement:
the InnerException might be null. Maybe log the message first and then if you really want the whole error you can use this method `public static string ExceptionToString(Exception exception)` in Util.cs.

Reply:
are you sure it can be null? 
if that was true, the original code would be broken
i tried to be careful and use the same assumptions as the original code
so if it was not broken, my new code should not be broken as well and vice versa

so looking at TargetInvocationException it seems that the inner exception is set (always?)
so then there are just 3 cases, firefilters which uses Exception
and sendmessage that uses AggregateException
that AggregateException in MSDN is also said to (always?) have inner exception set, that's comes from Task.Wait

so FireFilters remain, there i think you are right, that uses Exception and calls unknown code, so we can't guarantee anything about exception coming out, i'll fix FireFilters handler then, tx

Reply:
Thanks for investigating this mate.